src/storage/v2/property_store.cpp

```diff
@@ -520,6 +591,69 @@ std::optional<uint64_t> DecodeTemporalDataSize(Reader &reader) {
   return temporal_data_size;
 }
 
+std::optional<ZonedTemporalData> DecodeZonedTemporalData(Reader &reader) {
+  auto metadata = reader.ReadMetadata();
+
+  if (!metadata ||
+      (metadata->type != Type::ZONED_TEMPORAL_DATA && metadata->type != Type::OFFSET_ZONED_TEMPORAL_DATA)) {
+    return std::nullopt;
+  }
+
+  auto type_value = reader.ReadUint(metadata->id_size);
+  if (!type_value) return std::nullopt;
+
+  auto microseconds_value = reader.ReadInt(metadata->payload_size);
+  if (!microseconds_value) return std::nullopt;
+
+  if (metadata->type == Type::ZONED_TEMPORAL_DATA) {
+    auto tz_str_length = reader.ReadUint(TZ_NAME_LENGTH_SIZE);
```

Comment:
Maybe we can have a function in Reader class to read timezone in this format? The same as we have `ReadTimezoneOffset`

Reply:
Or even better we can incapsulate logic of types in one function and return `utils::Timezone`.

Reply:
Agree that encapsulation is a good idea, will do.